controllers/machineset_controller_test.go

```diff
@@ -709,3 +738,69 @@ func newMachineSet(name, cluster string) *clusterv1.MachineSet {
 		},
 	}
 }
+
+func TestMachineSetReconcile_MachinesCreatedConditionFalseOnBadInfraRef(t *testing.T) {
```

Comment:
Would it be possible to add more test cases?

Reply:
@sbueringer I added a couple of new test cases for the `ResizedCondition` Condition.